perceval/simulators/simulator.py

```diff
@@ -408,28 +415,45 @@ def probs_svd(self, input_dist: SVDistribution, progress_callback: Callable | No
             * physical_perf is the performance computed from the detected photon filter
             * logical_perf is the performance computed from the post-selection
         """
+        self._reset_perf()
+        min_detected_photons = self._min_detected_photons_filter
+        if self._heralds:
+            self._setup_mask_from_heralds()
+            min_detected_photons = max(min_detected_photons, sum(self._heralds.values()))
```

Comment:
I think this should raise an Exception or a warning (except if this is for retrocompatibility). We can't know if the given `min_detected_photons` is lower because the user forgot to count the heralds or if they forgot some photons

Reply:
Even if the user forgot to count some photons, increasing the `min_detected_photons` value will give the same correct results, without simulating input states that have to few. Still, you're right, I can show a user warning in this case.